src/grappler/ngraph_optimizer.cc

```diff
@@ -0,0 +1,188 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+#include "ngraph_optimizer.h"
+
+#include "tensorflow/core/framework/attr_value.pb.h"
+#include "tensorflow/core/framework/node_def.pb.h"
+#include "tensorflow/core/grappler/clusters/cluster.h"
+#include "tensorflow/core/grappler/grappler_item.h"
+#include "tensorflow/core/grappler/op_types.h"
+#include "tensorflow/core/grappler/optimizers/custom_graph_optimizer_registry.h"
+#include "tensorflow/core/grappler/utils.h"
+#include "tensorflow/core/platform/protobuf.h"
+
+#include <iomanip>
+#if defined NGRAPH_DISTRIBUTED
+#include "ngraph/distributed.hpp"
+#endif
+
+#include <iostream>
+
+using namespace std;
+
+namespace tensorflow {
+namespace ngraph_bridge {
+
+Status NgraphOptimizer::Optimize(tensorflow::grappler::Cluster* cluster,
+                                 const tensorflow::grappler::GrapplerItem& item,
+                                 GraphDef* output) {
+  NGRAPH_VLOG(1) << "Here at NgraphOptimizer ";
+
+  // Convert the GraphDef to Graph
+  GraphConstructorOptions opts;
+  opts.allow_internal_ops = true;
+  opts.expect_device_spec = true;
+  Graph graph(OpRegistry::Global());
+  TF_RETURN_IF_ERROR(ConvertGraphDefToGraph(opts, item.graph, &graph));
+
+  // Get the nodes to be skipped
+  std::vector<string> skip_these_nodes;
+  for (const string& f : item.fetch) {
+    NGRAPH_VLOG(5) << "Skip fetch node: " << f;
+    int pos = f.find(":");
+    skip_these_nodes.push_back(f.substr(0, pos));
+  }
+
+  //
+  // Variable capture: Part that replaces all instances of VariableV2 with the
+  // NGraphVariable op. Making this replacement allows us to substitute in a
+  // kernel that tracks the freshness of variables (invalidating freshness when
+  // the reference is handed off to an "untrusted" op).
+  //
+
+  // For filename generation purposes, grab a fresh index. This is just an
+  // arbitrary integer to avoid filename collisions resulting from subsequent
+  // runs of this pass.
+  int idx = FreshIndex();
+
+  // If requested, dump pre-capture graphs.
+  if (DumpPrecaptureGraphs()) {
+    DumpGraphs(graph, idx, "precapture", "Pre-Capture Graph");
+  }
+
+  // If ngraph is disabled via ngraph_bridge api or NGRAPH_TF_DISABLE is set
+  // we will not do anything; all subsequent
+  // passes become a no-op.
+  if (config::IsEnabled() == false ||
+      std::getenv("NGRAPH_TF_DISABLE") != nullptr) {
+    return Status::OK();
```

Comment:
Add a VLOG(0) indicating that NGRAPH is disabled.

Reply:
Done